syft/pkg/cataloger/python/package.go

```diff
@@ -55,13 +56,29 @@ func newPackageForRequirementsWithMetadata(name, version string, metadata pkg.Py
 	return p
 }
 
-func newPackageForPackage(m parsedData, sources ...file.Location) pkg.Package {
+func newPackageForPackage(resolver file.Resolver, m parsedData, sources ...file.Location) pkg.Package {
+	var licenseSet pkg.LicenseSet
+	if m.Licenses != "" {
+		licenseSet = pkg.NewLicenseSet(pkg.NewLicensesFromLocation(m.LicenseLocation, m.Licenses)...)
+	} else if m.LicenseLocation.Path() != "" {
+		// If we have a license file then resolve and parse it
+		found, _ := resolver.FilesByPath(m.LicenseLocation.Path())
```

Comment:
can you add `log.Trace` calls to show when there are errors here?
For instance
```golang
log.WithFields("error", err).Tracef("unable to fetch python license path %q", path)
```

Similar comment for when parsing fails a few lines below.

Reply:
Yep done